c/meterpreter/source/extensions/stdapi/server/net/config/arp.c

```diff
@@ -48,8 +45,27 @@ DWORD get_arp_table(Remote *remote, Packet *response)
 					arp[1].buffer        = (PUCHAR)pIpNetTable->table[i].bPhysAddr;
 
 					arp[2].header.type   = TLV_TYPE_MAC_NAME;
-					arp[2].header.length = (DWORD)strlen(interface_index) + 1;
-					arp[2].buffer        = (PUCHAR)interface_index;
+
+					BOOL has_description = FALSE;
+					MIB_IFROW iface;
+					iface.dwIndex = pIpNetTable->table[i].dwIndex;
+					result = GetIfEntry(&iface);
+					if (result == NO_ERROR)
+					{
+						if (iface.bDescr)
+						{
+							arp[2].header.length = (DWORD)strlen(iface.bDescr) + 1;
+							arp[2].buffer = (PUCHAR)iface.bDescr;
+							has_description = TRUE;
+						}
+					}
+
+					if (!has_description) {
+						char interface_index[10];
+						sprintf_s(interface_index, sizeof(interface_index), "%d", pIpNetTable->table[i].dwIndex);
+						arp[2].header.length = (DWORD)strlen(interface_index) + 1;
+						arp[2].buffer        = (PUCHAR)interface_index;
+					}

```

Comment:
You should be able to get rid of the `has_description` by tweaking the logic a bit.

Also, it's probably a good idea to switch the definition of `iface` so the `dwIndex` is initialized which causes the other fields to be zero'ed out.

```suggestion
					MIB_IFROW iface = { .dwIndex = pIpNetTable->table[i].dwIndex };
					result = GetIfEntry(&iface);
					if ((result == NO_ERROR) && (iface.bDescr)) {
						arp[2].header.length = (DWORD)strlen(iface.bDescr) + 1;
						arp[2].buffer = (PUCHAR)iface.bDescr;
					}
					else {
						char interface_index[10];
						sprintf_s(interface_index, sizeof(interface_index), "%d", pIpNetTable->table[i].dwIndex);
						arp[2].header.length = (DWORD)strlen(interface_index) + 1;
						arp[2].buffer        = (PUCHAR)interface_index;
					}

```